milatools/cli/remote.py

```diff
@@ -78,11 +79,16 @@ def get_first_node_name(node_names_out: str) -> str:
 class Remote:
     def __init__(self, hostname, connection=None, transforms=(), keepalive=60):
         self.hostname = hostname
-        if connection is None:
-            connection = Connection(hostname)
-            if keepalive:
-                connection.open()
-                connection.transport.set_keepalive(keepalive)
+        try:
+            if connection is None:
+                connection = Connection(hostname)
+                if keepalive:
+                    connection.open()
+                    connection.transport.set_keepalive(keepalive)
```

Comment:
I'm not sure the SSHException will occur here if keepalive is not true. I think the connection object is lazy, so it may only open the connection on the first command we try to run. I don't remember if we ever set keepalive=0, so it may not matter in practice, but it's something to keep in mind.